From 512c83eefc5d44f12b7b7a5f70869f8df70ca739 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Tue, 2 Apr 2013 12:59:09 -0700
Subject: [PATCH 53/96] Compatibility issue CDH 4.1/4.2 missing HTable constructors

Reason: Compatibility
Author: Matteo Bertozzi
Ref: CDH-10221
(cherry picked from commit 1aa0df2b5b055fa2ef9af6668ca22d05d598ede1)
---
 .../org/apache/hadoop/hbase/client/HTable.java     |   37 ++++++++++++++++++++
 1 files changed, 37 insertions(+), 0 deletions(-)

diff --git a/src/main/java/org/apache/hadoop/hbase/client/HTable.java b/src/main/java/org/apache/hadoop/hbase/client/HTable.java
index bc5f6e3..e00ebe1 100644
--- a/src/main/java/org/apache/hadoop/hbase/client/HTable.java
+++ b/src/main/java/org/apache/hadoop/hbase/client/HTable.java
@@ -120,6 +120,43 @@ public class HTable implements HTableInterface {
 
   /**
    * Creates an object to access a HBase table.
+   * Internally it creates a new instance of {@link Configuration} and a new
+   * client to zookeeper as well as other resources.  It also comes up with
+   * a fresh view of the cluster and must do discovery from scratch of region
+   * locations; i.e. it will not make use of already-cached region locations if
+   * available. Use only when being quick and dirty.
+   * @throws IOException if a remote or network exception occurs
+   * @deprecated use {@link #HTable(Configuration, String)}
+   */
+  @Deprecated
+  public HTable(final String tableName)
+  throws IOException {
+    this(HBaseConfiguration.create(), Bytes.toBytes(tableName));
+    LOG.warn("This constructor HTable(byte[]) is deprecated and it will be removed. " +
+             "Please use HTable(Configuration, byte[]) to construct an HTable");
+  }
+
+  /**
+   * Creates an object to access a HBase table.
+   * Internally it creates a new instance of {@link Configuration} and a new
+   * client to zookeeper as well as other resources.  It also comes up with
+   * a fresh view of the cluster and must do discovery from scratch of region
+   * locations; i.e. it will not make use of already-cached region locations if
+   * available. Use only when being quick and dirty.
+   * @param tableName Name of the table.
+   * @throws IOException if a remote or network exception occurs
+   * @deprecated use {@link #HTable(Configuration, byte[])}
+   */
+  @Deprecated
+  public HTable(final byte [] tableName)
+  throws IOException {
+    this(HBaseConfiguration.create(), tableName);
+    LOG.warn("This constructor HTable(byte[]) is deprecated and it will be removed. " +
+             "Please use HTable(Configuration, byte[]) to construct an HTable");
+  }
+
+  /**
+   * Creates an object to access a HBase table.
    * Shares zookeeper connection and other resources with other HTable instances
    * created with the same <code>conf</code> instance.  Uses already-populated
    * region cache if one is available, populated by any other HTable instances
-- 
1.7.0.4

